designer/pydesigner.py

```diff
@@ -711,7 +715,35 @@
                 akc_out = img.akcoutliers()
                 img.akccorrect(akc_out)
             mk, rk, ak, kfa, mkt, trace = img.extractDKI()
+            if args.median:
+                if args.mask:
+                    brainmaskPath = op.join(outpath, 'brain_mask.nii')
+                    import nibabel as nib
+                    maskHDR = nib.load(brainmaskPath)
+                    mask = np.array(maskHDR.dataobj)
+                else:
+                    mask = None
+                filt = dp.medianFilter(img=mk,
+                                       brainmask=mask,
+                                       tissueth=0.6,
+                                       th=0.5,
+                                       sz=3,
+                                       conn='face',
+                                       bias='rand')
+                mk = filt.apply(mk, weight=1)
+                rk = filt.apply(rk, weight=1)
+                ak = filt.apply(ak, weight=1)
+                kfa = filt.apply(kfa, weight=1)
+                mkt = filt.apply(mkt, weight=1)
+                md = filt.apply(md, weight=0.5)
```

Comment:
Explain why the weightings are different in a comment, or use some constants, e.g.,
```
DKI_WEIGHT = 1
DTI_WEIGHT = 0.5
```